impl/src/main/java/org/jboss/weld/bean/proxy/InterceptedSubclassFactory.java

```diff
@@ -437,4 +450,61 @@ private void createDelegateMethod(ClassFile proxyClassType, Method method, Metho
         createDelegateToSuper(delegatingMethod, methodInformation);
     }
 
-}
+    private static class BridgeMethod {
```

Comment:
I know it more or less doesn't matter, but wouldn't it be a tad bit nicer to have this as a separate class file which `extends MethodSignature`?

Reply:
We can make it separate once it is usable outside `InterceptedSubclassFactory`. Also extending `MethodSignature` does not bring any value here. And by the way, it's not a "method signature" as it contains the return type ;-)

Reply:
That's why it would be an 'extended' method signature, right? :-P